Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -299,7 +292,8 @@ private struct _BSONKeyedDecodingContainer<K: CodingKey> : KeyedDecodingContaine
     private func decodeNumber<T: CodableNumber>(_ type: T.Type, forKey key: Key) throws -> T {
         let entry = try getValue(forKey: key)
         return try self.decoder.with(pushedKey: key) {
-            guard let value = try decoder.unboxNumber(entry, as: type) else {
+            let value = try decoder.unboxNumber(entry, as: type)
+            guard !(value is NSNull) else {
```

Comment:
is it possible that `unboxNumber` ever returns an `NSNull`? I think it will only return something of the provided type, which must be a `CodableNumber`

Reply:
ya, l agree, not sure why that was returning an optional. oops